### ISSUE TITLE
bug: alacritty v0.13 and greater continues to create YAML file

### ISSUE THREAD

**Issue by scallaway** [nix-community/home-manager#5215]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

Installing the unstable version of Alacritty through `home-manager`, I get version v0.13.2. As per https://github.com/nix-community/home-manager/pull/4817, the configuration file that's generated based on my `settings` is `~/.config/alacritty/alacritty.yml`.

When running the version specified, a warning is placed at the bottom of the screen telling me to run `alacritty migrate` as the YAML file is deprecated.

I have had a look over the `alacritty.nix` file here, although I'm still very green to NixOS and the language itself, so I'm not able to offer any pointers there.

### Maintainer CC

@r-vdp @ncfavier

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 6.1.82, NixOS, 23.11 (Tapir), 23.11.20240328.219951b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"home-manager-23.11.tar.gz, nixos-23.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by r-vdp** [comment]
The info you posted says that you run nixos 23.11 but AFAICT, nixpkgs 23.11 is still on alacritty 12, which uses yaml. Did you use an override or such to update alacritty on your system?

If you use the same versions of both HM and nixos, this should not happen, if I'm reading the code and versions correctly. But we could consider backporting the change that makes the module work for alacritty 13+.

@rycee should we backport #4817 to release-23.11?

**Comment by scallaway** [comment]
> The info you posted says that you run nixos 23.11 but AFAICT, nixpkgs 23.11 is still on alacritty 12, which uses yaml. Did you use an override or such to update alacritty on your system?

Yeah so I'm selecting the `alacritty` package out of the unstable branch manually (specifying with the `package` property in `programs.alacritty`).

I'd love for these changes to be backported to the stable channel, but I'm concerned it's still not working on the unstable branch as-is.

**Comment by r-vdp** [comment]
I'm using alacritty on unstable without issues. The problem is that you're using the HM module from the release-23.11 branch of HM still, which doesn't know about the new toml format.

**Comment by scallaway** [comment]
Right that makes sense, I didn't realise that was also a requirement.

I'm guessing there isn't a way to selectively run the HM module on the unstable branch for alacritty alone?

**Comment by scallaway** [comment]
Fwiw, I've swapped my device over to using the unstable branch of HM and can confirm that fixes the issue as you outlined.

Would still love for this to be backported to stable though!

**Comment by csumpasd** [comment]
I'm having the same issue (on nix-darwin + home-manager), but would prefer not to use the unstable nixpkgs globally. Is there a way to selectively use the unstable home-manager for configuring alcritty? I didn't find anything about this online.

**Comment by csumpasd** [comment]
alright, thank you!

**Comment by teto** [comment]
looks fixed ?
